kotlin-native/build-tools/src/main/kotlin/org/jetbrains/kotlin/Utils.kt

```diff
@@ -47,9 +83,47 @@ val Project.testOutputFramework
 val Project.testOutputExternal
     get() = (findProperty("testOutputExternal") as File).toString()
 
+val Project.cacheRedirectorEnabled
+    get() = findProperty("cacheRedirectorEnabled")?.toString()?.toBoolean() ?: false
+
+fun Project.redirectIfEnabled(url: String):String = if (cacheRedirectorEnabled) {
+    val base = URL(url)
+    "https://cache-redirector.jetbrains.com/${base.host}/${base.path}"
+} else
+    url
+
+val validPropertiesNames = listOf("kotlin.native.home",
+                                  "org.jetbrains.kotlin.native.home",
+                                  "konan.home")
+
 val Project.kotlinNativeDist
-    get() = this.rootProject.file(this.findProperty("org.jetbrains.kotlin.native.home")
-            ?: this.findProperty("konan.home") ?: "dist")
+    get() = rootProject.file(validPropertiesNames.firstOrNull{ hasProperty(it) }?.let{ findProperty(it) } ?: "dist")
```

Comment:
What about location of the `dist`? Should it be in the root project or in the `kotlin-native` dir?

Reply:
I don't know, I haven't find any issues with enabled and disabled K/N neither in rr-tests nor local tests. if absolutely it's mandatory to separate local dists I'd suggest kotlin-native-dist or dist-kotlin-native, it's much easier to navigate from the root of the project.

Reply:
I'm afraid of any possible issues with packing distribution during release cycle. If for some reason Kotlin's dist will be build, and we build our's also, both will be packed into the zips/tars.

Reply:
This scenario possible if kotlin.native is enabled, and we somehow missed porting release procedures. to not forget this part let design in scope of releasing procedure porting (KTI-482 develop k/n releasing procedure in migration period)